python/config_settings/config_settings.bzl

```diff
@@ -59,14 +62,39 @@ def construct_config_settings(name, python_versions):
         )
         matches_minor_version_names = [equals_minor_version_name]
 
+        default_micro_version = MINOR_MAPPING.get(minor_version)
+
         for micro_version in micro_versions:
             is_micro_version_name = "is_python_" + micro_version
+            if not default_micro_version:
+                native.config_setting(
+                    name = is_micro_version_name,
+                    flag_values = {":python_version": micro_version},
+                    visibility = ["//visibility:public"],
+                )
+                matches_minor_version_names.append(is_micro_version_name)
+                continue
+
+            # Ensure that is_python_3.9.8 is matched if python_version is set
+            # to 3.9 if MINOR_MAPPING points to 3.9.8
+            equals_micro_name = "_python_version_flag_equals_" + micro_version
             native.config_setting(
-                name = is_micro_version_name,
+                name = equals_micro_name,
                 flag_values = {":python_version": micro_version},
+            )
+            selects.config_setting_group(
```

Comment:
```suggestion
            # An alias pointing to an underscore-prefixed config_setting_group
            # is used because config_setting_group creates
            # `is_{minor}_N` targets, which are easily confused with the
            # `is_{minor}.{micro}` (dot) targets.
            selects.config_setting_group(
```